src/main/java/org/akhq/controllers/TopicController.java

```diff
@@ -320,6 +320,30 @@ public Record deleteRecordApi(String cluster, String topicName, Integer partitio
             });
     }
 
+    @Secured(Role.ROLE_TOPIC_DATA_READ)
+    @Get("api/{cluster}/topic/{topicName}/data/record")
+    @Operation(tags = {"topic data"}, summary = "Get a single record by partition and offset")
+    public ResultNextList<Record> record(
```

Comment:
I would go to path parameter instead of querystring since the argument are mandatory for offset and partition
and change 
```java
Optional<String> after,
Optional<Integer> partition
```
to 
```java
String offset,
Integer partition
```

Reply:
Maybe 
```
Integer offset,
Integer partition
```

We only pass the offset number, and do the join for dataSearchOptions after. 

In RecordRepository.getFirstOffsetForSortOldest the 
`first = options.after.get(partition.getId()) + 1;`  
will never show offset 0